test/react/hooks/use-typing.test.tsx

```diff
@@ -0,0 +1,212 @@
+import { ConnectionLifecycle, Room, RoomLifecycle, TypingListener } from '@ably/chat';
+import { act, renderHook } from '@testing-library/react';
```

Comment:
Can we get an integration test that runs end-end using multiple clients and a sandbox?